pkg/ioutils/readers.go

```diff
@@ -189,6 +190,9 @@ func (r *bufReader) drain() {
 		reuseCount++
 		r.wait.Signal()
 		r.Unlock()
+		//allow or force Go scheduler to switch context, without explicitly
+		//forcing this will make it hang when using gccgo implementation
+		runtime.Gosched()
```

Comment:
can we do this only when it's built with gccgo? I'm thinking that having this behind a build tag could be a good idea.